### ISSUE TITLE
`isinstance(...) is True` type guard

### ISSUE THREAD

**Issue by nihaals** [microsoft/pyright#2154]
**Is your feature request related to a problem? Please describe.**
Some people prefer using `if ... is True` or `if ... is False` over `if ...` or `if not ...`. This is not currently supported.

**Describe the solution you'd like**
`if ... is True` and `if ... is False` to have the same effect as `if ...` and `if not ...`.

**Additional context**
Example:
```py
def isinstance_true(arg):
    if isinstance(arg, str) is True:
        reveal_type(arg)  # Type of "arg" is "Unknown"

def isinstance_implicit_true(arg):
    if isinstance(arg, str):
        reveal_type(arg)  # Type of "arg" is "str"

def isinstance_false(arg):
    if isinstance(arg, str) is False:
        pass
    else:
        reveal_type(arg)  # Type of "arg" is "Unknown"

def isinstance_implicit_false(arg):
    if not isinstance(arg, str):
        pass
    else:
        reveal_type(arg)  # Type of "arg" is "str"
```

Output:
```
$ pyright --outputjson test.py
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/tmp.3JLrAbOdWe/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
{
    "version": "1.1.159",
    "time": "1628198387973",
    "generalDiagnostics": [
        {
            "file": "/tmp/tmp.3JLrAbOdWe/test.py",
            "severity": "information",
            "message": "Type of \"arg\" is \"Unknown\"",
            "range": {
                "start": {
                    "line": 2,
                    "character": 20
                },
                "end": {
                    "line": 2,
                    "character": 23
                }
            }
        },
        {
            "file": "/tmp/tmp.3JLrAbOdWe/test.py",
            "severity": "information",
            "message": "Type of \"arg\" is \"str\"",
            "range": {
                "start": {
                    "line": 6,
                    "character": 20
                },
                "end": {
                    "line": 6,
                    "character": 23
                }
            }
        },
        {
            "file": "/tmp/tmp.3JLrAbOdWe/test.py",
            "severity": "information",
            "message": "Type of \"arg\" is \"Unknown\"",
            "range": {
                "start": {
                    "line": 12,
                    "character": 20
                },
                "end": {
                    "line": 12,
                    "character": 23
                }
            }
        },
        {
            "file": "/tmp/tmp.3JLrAbOdWe/test.py",
            "severity": "information",
            "message": "Type of \"arg\" is \"str\"",
            "range": {
                "start": {
                    "line": 18,
                    "character": 20
                },
                "end": {
                    "line": 18,
                    "character": 23
                }
            }
        }
    ],
    "summary": {
        "filesAnalyzed": 14,
        "errorCount": 0,
        "warningCount": 0,
        "informationCount": 4,
        "timeInSec": 0.229
    }
}
```

**Comment by erictraut** [comment]
This isn't one of the type guard patterns currently supported by pyright. For a complete list of supported patterns, refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/type-concepts.md#type-guards).

Each new pattern requires specialized code that adds complexity and potentially slows down analysis. We do occasionally add new support for new patterns but only in cases where we get a signal that it's a commonly-used idiom. We use two ways to gather that signal: 1) upvotes on issues, and 2) support for the pattern in other Python type checkers like mypy. In this particular case, I've never personally seen this pattern used in Python code. Mypy also doesn't support this pattern, and I don't see any feature requests in the mypy issue tracker requesting this feature. So I don't think this is a common enough pattern for us to support at this time. We can revisit this decision in the future if this issue receives enough upvotes.

In the meantime, there's a trivially easy workaround (that arguably makes the code more readable): just delete the `is True` or `is False`.
